scripts/CEE/collect-verbose-ca-logs/script.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# Set variables
+GREEN="\033[32m"
+RESET="\033[0m"
+NAMESPACE="openshift-machine-api"
+CA_NAME="default"
+
+# Get current logVerbosity value
+current_log_verbosity=$(oc get ca $CA_NAME -o jsonpath='{.spec.logVerbosity}')
+
+# Check if current_log_verbosity is empty
+if [ -z "$current_log_verbosity" ]; then
+  echo "Failed to retrieve current logVerbosity value."
+  exit 1
+fi
+
+echo
+echo "CURRENT LOG VERBOSITY = $current_log_verbosity"
+echo
+
+# Update logVerbosity to 6
+oc patch ca $CA_NAME --type='json' -p='[{"op": "replace", "path": "/spec/logVerbosity", "value": 6}]'
```

Comment:
The script called the patch operation twice on the resource, and it will trigger the redeploy twice as well.
I feel it should not be the regular task.

Reply:
I assume this is to turn on the verbose to print more info and then turn off the verbose to make it back to normal.
https://access.redhat.com/solutions/6821651
https://access.redhat.com/solutions/6172162
https://docs.openshift.com/container-platform/4.12/machine_management/applying-autoscaling.html#cluster-autoscaler-cr_applying-autoscaling

@kkulshreshth  if that's the case, it would be great if you can share an example output for reviewers' reference.

Reply:
@feichashao You are absolutely right on this. This is to reset the logVerbosity. @bmeng you are right that it will redeploy the pod. Mostly the customers keep logVerbosity to 1 and CA logs with verbosity 1 is literally of no use.

For example, consider the case where a customer has underutilized nodes. With logVerbosity 1, the CA pod prints:
`I0605 09:26:35.309013       1 eligibility.go:102] Scale-down calculation: ignoring 6 nodes unremovable in the last 5m0s
I0605 09:26:35.907516       1 legacy.go:296] No candidates for scale down`

With logVerbosity 6, we get a detailed log highlighting the root cause behind no scale down:
`I0606 07:50:35.265451       1 klogx.go:87] Node osdmgmtdrprd01-c54h8-worker-16-b-q9c4v - cpu utilization 0.417810
I0606 07:50:35.266018       1 cluster.go:155] osdmgmtdrprd01-c54h8-worker-16-b-q9c4v for removal
I0606 07:50:35.266031       1 cluster.go:164] node osdmgmtdrprd01-c54h8-worker-16-b-q9c4v cannot be removed: pod with local storage present: openshift-gitops-repo-server-74694cd9d6-25l52`

**MCS will run this script only after taking customer's consent and only CRU members should be able to run this script.** Additionally, we will integrate this script with our troubleshooting CLI tool (inceptor).

Let me know if you have further questions.

Reply:
thanks for explaining this